aexpect/session_ops.py

```diff
@@ -0,0 +1,598 @@
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the size of ``path``
+    :rtype: int
+    """
+    return int(stat(session, path, fmt="%s"))
+
+
+###############################################################################
+# test(1)
+###############################################################################
+
+def test(session, path, flag):
+    """
+    Wrapper for ``test``.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to test
+    :param str flag: Flag passed to test
+    :returns: status of test(1) on ``path``
+    :rtype: int
+    """
+    if not isinstance(flag, str) or flag[0] != "-":
+        raise RuntimeError("%s is not a valid flag string for test(1)" % flag)
+
+    return int(session.cmd_status(r"test %s '%s'" % (flag, path)))
+
+
+def is_directory(session, path):
+    """
+    Check if a directory exists on a given guest.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the directory to check
+    :returns: whether the directory exists
+    :rtype: bool
+    """
+    return test(session, path, flag="-d") == 0
+
+
+def is_regular_file(session, path):
+    """
+    Check if a regular file exists on a given guest.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the regular file to check
+    :returns: whether the regular file exists
+    :rtype: bool
+    """
+    return test(session, path, flag="-f") == 0
+
+
+def is_nonzero_size_file(session, path):
+    """
+    Check if a regular file exists on a given guest.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the regular file to check
+    :returns: whether the regular file exists and is nonzero size
+    :rtype: bool
+    """
+    return test(session, path, flag="-s") == 0
+
+
+###############################################################################
+# file ops
+###############################################################################
+
+def hash_file(session, filename, method='md5'):
+    """
+    Calculate hash of given file from a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str filename: full path of file that should be hashed
+    :param str method: method for hashing
+    :returns: hash of file, a hex number
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if hash command failed (e.g. file not
+             found) or resulting output does not have expected format
+    :raises: :py:class:`ValueError` if given method is not (yet) supported
+    """
+    if method == 'md5':
+        cmd = 'md5sum'
+        expect_len = 32
+    else:
+        raise ValueError('only method "md5" supported yet')
```

Comment:
For now, to reduce the complexity, I would suggest, something like:

```python
if method not in ['md5']:
    raise ValueError('only method "md5" supported yet')

cmd = 'md5sum'
expect_len = 32
```

But when we add new methods, my suggestion is to break the method into specific methods.

Reply:
Both hands agreed, thanks for spotting this.